src/mutate/deleteAndMergeSplitNodes.ts

```diff
@@ -134,10 +135,42 @@ export function deleteAndMergeSplitNodes(
         // Same as above, merge nodes manually if there exists an open slice with mergeable content.
         // Compared to deleting an end token however, the merged block node is set as deleted. This is due to
         // ProseMirror node semantics as start tokens are considered to contain the actual node itself.
-        const mergeEndNode =
-          startTokenDeleted && openEnd > 0 && depth === openEnd && mergeContent && mergeContent.size
+        const mergeEndNode = startTokenDeleted && openEnd > 0 && depth === openEnd && mergeContent
 
-        if (mergeStartNode || mergeEndNode) {
+        const mergeEndNodeNotEmpty = mergeEndNode && mergeContent.size
+        if (mergeEndNode && !mergeEndNodeNotEmpty && gap) {
+          /* Here to support a case when we lift a node from a multichild parent - meaning that we take out a node from another node (parent)
+            which would result in gluing of the parent at the position of harvesting, e.g. lifting a <li> out of <ul> with multiple <li> will result in replace around step
+            that takes out that gap (e.g. first <li></li>) and glues a start of the <ul> without such content to the old ul.
+            such operation will not be recognised as 'nodeCompletelyDeleted' as all nodes will have openEnd, which means that they are going to be glued
+            to their older versions at the place of harvesting of lifted nodes. Such condition has to be captured and harvested node reinserted as deleted
+          */
+          if (gap.start < gap.end && gap.insert === 0 && gap.end === to && !node.isText) {
```

Comment:
can we move this to a function(isLiftedNode) with comment up

Reply:
done